crates/bevy_asset/src/lib.rs

```diff
@@ -452,52 +453,79 @@ pub trait AsAssetId: Component {
 /// This trait defines how to visit the dependencies of an asset.
 /// For example, a 3D model might require both textures and meshes to be loaded.
 ///
+/// The visitation function can return [`ControlFlow::Break`] to terminate visitation early.
+///
 /// Note that this trait is automatically implemented when deriving [`Asset`].
 pub trait VisitAssetDependencies {
-    fn visit_dependencies(&self, visit: &mut impl FnMut(UntypedAssetId));
+    fn visit_dependencies(
+        &self,
+        visit: &mut impl FnMut(UntypedAssetId) -> ControlFlow<UntypedAssetId>,
```

Comment:
I don't think I'm understanding how this control flow works. Let's say you have an asset with:
```
struct MyAsset(#[dependency] Vec<Handle<Mesh>>);
```

If for whatever reason one of those meshes returns ControlFlow::Break (because it was modified this frame), the entire `MyAsset` stops iterating, right?